stac_fastapi/pgstac/app.py

```diff
@@ -64,17 +83,30 @@
 else:
     items_get_request_model = ItemCollectionUri
 
-post_request_model = create_post_request_model(extensions, base_model=PgstacSearch)
-get_request_model = create_get_request_model(extensions)
+if any(isinstance(ext, CollectionSearchExtension) for ext in collection_extensions):
+    collections_get_request_model = create_request_model(
+        model_name="CollectionsGetRequest",
+        base_model=BaseCollectionSearchGetRequest,
+        extensions=extensions,
```

Comment:
Enable args from other extensions in the `/collections` GET request model

Reply:
should `extensions` should be `collection_extensions` ?

Reply:
also I don't think we need to `test` with isinstance, just a `if collection_extensions:` should be enough

Reply:
It needs to be `extensions` because want to add the parameters from other extensions (like `Filter`, `Sort`) to the collections GET request if those extensions are enabled. It's confusing but this only gets run **if** the collection search extension is enabled.

Reply:
so `collection_extensions` is not really used then? (sorry I'm trying to catch back after two weeks away from stac-fastapi-* 😓)

Reply:
ok so:
- we want to check if `collection_search` is enabled in `ENABLED_EXTENSIONS` env. 
- if enable, we need to construct a model based on `BaseCollectionSearchGetRequest` which we can extent with the other extensions

makes sense 👍 , maybe add some docstring reflecting ☝️ 

I feel splitting the to `extensions` and `collection_extensions` might not be needed. I think I'm the one who proposed it but maybe it's useless

Reply:
well looking at https://github.com/stac-utils/stac-fastapi/blob/4979a89b4dc1534645388f182046955986fee669/stac_fastapi/extensions/stac_fastapi/extensions/core/collection_search/collection_search.py#L20-L72 

it's a bit more complex, and I don't think we can simply add extension to the collection-search. Technically we would also need to change the conformances list if `sort` or `filter` were added to the collection-search

IMO, the application should implement a minimal solution (simple collection-search) but we can focus on the documentation to explain how to extend the collection-search to allow extensions

Reply:
I think adding extensions to the collection-search extension will be really useful (especially `free-text` and `filter`). I'll come up with something that can detect which conformance classes to add based on the enabled extensions. I agree that it is a little messy, so if this is another case where you would want to have users enable it via custom runtimes I guess that would work. I just think robust collection-search would be a really nice feature to have on by default!

Reply:
That looks like a much cleaner solution! I'll update this PR to skip adding extensions to the collection-search extension then follow up with a PR in stac-fastapi with this `classmethod`.

Reply:
I bumped the stac-fastapi requirement to `3.0.2` so we can use the new method - thanks for adding that @vincentsarago!